viam-cartographer.go

```diff
@@ -188,8 +180,8 @@ func New(
 
 // cartographerService is the structure of the slam service.
 type cartographerService struct {
-	generic.Unimplemented
-	name              string
+	resource.Named
+	resource.AlwaysRebuild
```

Comment:
TriviallyReconfigurable won't call close, and we need to call close to shut down the data process and carto_grpc_server